site-src/geps/gep-713.md

```diff
@@ -0,0 +1,409 @@
+
+This approach can be used to attach a default set of policies to all requests
+coming from a namespace. The example below shows a ServiceBinding defined in the
+producer namespace that would apply to all requests from within the same
+namespace or from other namespaces that did not have their own ServiceBindings
+defined.
+
+![Complex Service Binding Example](images/713-servicebinding-complex.png)
+
+#### Advantages
+* Works well for mesh and any use cases where requests don’t always transit
+  through Gateways and Routes.
+* Allows policies to apply to an entire namespace.
+* Provides very clear attachment of polices, routes, and more to a specific
+  Service.
+* Works well for ‘shrink-wrap application developers’ - the packaged app does
+  not need to know about hostnames or policies or have extensive templates.
+* Works well for ‘dynamic’ / programmatic creation of workloads ( Pods,etc - see
+  CertManager)
+* It is easy to understand what policy applies to a workload - by listing the
+  bindings in the namespace.
+
+#### Disadvantages
+* Unclear how this would work with an ingress model. If Gateways, Routes, and
+  Backends are all in different namespaces, and each of those namespaces has
+  different ServiceBindings applying different sets of policies, it’s difficult
+  to understand which policy would be applied.
+* Unclear if/how this would interact with existing the ingress focused policy
+  proposal described below. If both coexisted, would it be possible for a user
+  to understand which policies were being applied to their requests?
+* Route status could get confusing when Routes were referenced as a policy by
+  ServiceBinding
+* Introduces a new mesh specific resource.
+
+### 2. Attaching Policies for Ingress
+An earlier proposal for policy attachment in the Gateway API suggested adding
+policy references to each Resource. This works very naturally for Ingress use
+cases where all requests follow a path through Gateways, Routes, and Backends.
+Adding policy attachment at each level enables different roles to define
+defaults and allow overrides at different levels.
+
+![Simple Ingress Attachment Example](images/713-ingress-attachment.png)
+
+#### Advantages
+* Consistent policy attachment at each level
+* Clear which policies apply to each component
+* Naturally translates to hierarchical Ingress model with ability to delegate
+  policy decisions to different roles
+
+#### Disadvantages
+* Policy overrides could become complicated
+* At least initially, policy attachment on Service would have to rely on Service
+  annotations or references from policy to Service(s)
+* No way to attach policy to other resources such as namespace or ServiceImport
+* May be difficult to modify Routes and Services if other components/roles are
+  managing them (eg Knative)
+
+
+## Out of scope
+
+* Define all potential policies that may be attached to resources
+* Design the full structure and configuration of policies
```

Comment:
@robscott, have we considered having a GenericPolicy.gateway.networking.k8s.io resource which defines the API and allows for an arbitrary map[string]string?
Pros:
- For some tiny features, implementers can use that
- Implementers have a concrete knowledge of what is it that we are recommending
Cons:
- If this results in a wide-spread adoption, having type-less CRDs will result in more pain elsewhere in the community

Reply:
Yeah this is a really interesting idea. I think it's significantly better than annotations, but I'm concerned that it may pull away some implementations that would otherwise create CRDs (which I think results in the ideal UX). After chatting a bit more about this with you I've added this as an alternative in the GEP. It's really more of a potential variation but I wasn't sure where to put that.